### PR TITLE
Fastapi logging

### PR DESCRIPTION
**Pull request by hexfati**
Redirected Fastapi logging to file `fastapi_server.log` during normal operation mode and to `QTextEditLogger` during test mode

### CODE REVIEW

**Review by alessandropalla — APPROVED** [review]

**Comment by hexfati** [comment]
> in AnimalDetectionAndClassifiation mode while running GET on actuator i got following error in console:
> 
> 2024-11-12 16:50:51 INFO: Welcome to WADAS! 2024-11-12 16:51:02 INFO: Loading configuration from file... 2024-11-12 16:51:03 INFO: Configuration loaded from file C:/Users/stefa/Documents/config_1_1.yaml. 2024-11-12 16:51:26 INFO: Selected mode from dialog: Animal Detection and Classification Mode 2024-11-12 16:51:27 WARNING: No notification protocol set. 2024-11-12 16:51:28 INFO: initializing model... 2024-11-12 16:51:28 INFO: Initializing AI model for image detection and classification to auto... 2024-11-12 16:51:28 INFO: Initializing detection model to device auto... 2024-11-12 16:51:35 INFO: Loading classification model to device auto... 2024-11-12 16:51:44 INFO: Instantiating cameras... 2024-11-12 16:51:44 INFO: Instantiating FTPS server... 2024-11-12 16:51:44 INFO: Starting thread for FTPs Server... 2024-11-12 16:51:44 INFO: Instantiating HTTPS Actuator server... 2024-11-12 16:51:44 INFO: Starting thread for HTTPS Actuator Server with FastAPI... 2024-11-12 16:51:44 INFO: Starting thread for update actuators view... 2024-11-12 16:51:44 INFO: Ready for video stream from Camera(s)... 2024-11-12 16:52:41 ERROR: Exception in callback _ProactorBasePipeTransport._call_connection_lost(None) handle: <Handle _ProactorBasePipeTransport._call_connection_lost(None)> Traceback (most recent call last): File "C:\Users\stefa\anaconda3\envs\WADAS\lib\asyncio\events.py", line 80, in _run self._context.run(self._callback, *self._args) File "C:\Users\stefa\anaconda3\envs\WADAS\lib\asyncio\proactor_events.py", line 165, in _call_connection_lost self._sock.shutdown(socket.SHUT_RDWR) ConnectionResetError: [WinError 10054] Connessione in corso interrotta forzatamente dall'host remoto 2024-11-12 16:52:44 INFO: Connected remote actuator with ID: Actuator1 2024-11-12 16:52:50 ERROR: Exception in callback _ProactorBasePipeTransport._call_connection_lost(None) handle: <Handle _ProactorBasePipeTransport._call_connection_lost(None)> Traceback (most recent call last): File "C:\Users\stefa\anaconda3\envs\WADAS\lib\asyncio\events.py", line 80, in _run self._context.run(self._callback, *self._args) File "C:\Users\stefa\anaconda3\envs\WADAS\lib\asyncio\proactor_events.py", line 165, in _call_connection_lost self._sock.shutdown(socket.SHUT_RDWR) ConnectionResetError: [WinError 10054] Connessione in corso interrotta forzatamente dall'host remoto 2024-11-12 16:52:51 INFO: Connected remote actuator with ID: Actuator1
> 
> @hexfati Something might still be missing FastAPI logging redirect...

This traceback is not handled by fastapi logger. We can merge these changes about fastapi and then work separately on this other issue.

**Comment by stefanodellosa-personal** [comment]
Ok, let's track this in separate issue.

**Review by stefanodellosa-personal — APPROVED** [review]

File: wadas/domain/fastapi_actuator_server.py [hunk 1]
```diff
@@ -2,13 +2,34 @@
...
 
+def initialize_fastapi_logger():
+    """Method to initialize Fastapi server logger"""
+    handler = RotatingFileHandler(
+        os.path.join("log", "fastapi_server.log"), maxBytes=100000, backupCount=3
+    )
+    formatter = logging.Formatter("%(asctime)s %(levelname)s: %(message)s", "%Y-%m-%d %H:%M:%S")
+    handler.setFormatter(formatter)
+
+    logger_names = ["uvicorn", "uvicorn.error", "uvicorn.access"]
```
Comment (cdimauro) [inline wadas/domain/fastapi_actuator_server.py:22]: Replace [] with ().

File: wadas/ui/configure_actuators_dialog.py [hunk 2]
```diff
@@ -362,24 +361,29 @@ def start_actuator_server(self):
...
+        logger_names = ["uvicorn", "uvicorn.error", "uvicorn.access"]
         log_textbox = QTextEditLogger(self.ui.plainTextEdit_test_server_log)
-        logger.setLevel(logging.DEBUG)
-        logger.addHandler(log_textbox)
+        for logger_name in logger_names:
+            server_logger = logging.getLogger(logger_name)
+            for handler in server_logger.handlers[:]:
+                server_logger.removeHandler(handler)
+            server_logger.addHandler(log_textbox)
+            server_logger.propagate = False
```
Comment (cdimauro) [inline wadas/ui/configure_actuators_dialog.py:389]: This is very similar to L22-30 in fastapi_actuator_server.py. You may create a common function that just takes a couple of arguments: the handler to add and an optional debug level.

**Pull request merged**